libafl/src/feedbacks/new_hash_feebdack.rs

```diff
@@ -0,0 +1,82 @@
+//! The NewHashFeebdacks uses the backtrace hash and a hashset to only keep novel cases
+
+use hashbrown::HashSet;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    bolts::tuples::Named,
+    events::EventFirer,
+    executors::ExitKind,
+    feedbacks::Feedback,
+    inputs::Input,
+    observers::{ObserversTuple, StacktraceObserver},
+    state::HasClientPerfMonitor,
+    Error,
+};
+
+/// A [`NewHashFeebdack`] maintains a hashset of already seen stacktraces and considers interesting unseen ones
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct NewHashFeebdack {
+    observer_name: String,
+    hashset: HashSet<u64>,
+}
+
+impl<I, S> Feedback<I, S> for NewHashFeebdack
+where
+    I: Input,
+    S: HasClientPerfMonitor,
+{
+    fn is_interesting<EM, OT>(
+        &mut self,
+        _state: &mut S,
+        _manager: &mut EM,
+        _input: &I,
+        observers: &OT,
+        _exit_kind: &ExitKind,
+    ) -> Result<bool, Error>
+    where
+        EM: EventFirer<I, S>,
+        OT: ObserversTuple<I, S>,
+    {
+        let observer = observers
+            .match_name::<StacktraceObserver>(&self.observer_name)
+            .expect("A NewHashFeebdack needs a StacktraceObserver");
+
+        match observer.hash() {
+            Some(hash) => {
+                if self.hashset.contains(hash) {
+                    Ok(false)
+                } else {
+                    self.hashset.insert(*hash);
+                    Ok(true)
+                }
+            }
+            // Something's wrong if we get here
+            None => Ok(false),
+        }
+    }
+}
+
+impl Named for NewHashFeebdack {
+    #[inline]
+    fn name(&self) -> &str {
+        "NewHashFeebdack"
```

Comment:
If this name is static we can only have one of them at any point in time. Rather, it should be a field in the struct that can be set on creation (I think)

Reply:
no it is ok, this is not relevant apart from stats without a feedback state

Reply:
For example, if you want to do differential fuzzing and have two feedbacks in the executor, or if you hash something else than additionally and want to reuse the same feedback